setup.py

```diff
@@ -57,7 +57,8 @@
                   'mock',
                   'pytest-mpl',
                   'pytest-cov',
-                  'pytest < 4'],
+                  'pytest',
+                  'contextlib2'],
```

Comment:
contextlib2 is used to create dummy contexts that are used as dynamic failure catches in the new tests.  This is because we can no longer mark certain parametrizations as failure modes.